plugins/woocommerce/client/legacy/js/admin/wc-shipping-zone-methods.js

```diff
@@ -197,16 +198,33 @@
 						model   = view.model,
 						methods   = _.indexBy( model.get( 'methods' ), 'instance_id' ),
 						changes = {},
-						instance_id = $( this ).closest('tr').data('id');
+						instance_id = $( this ).closest( 'tr' ).data( 'id' );
 
 					event.preventDefault();
-
-					delete methods[ instance_id ];
-					changes.methods = changes.methods || { methods : {} };
-					changes.methods[ instance_id ] = _.extend( changes.methods[ instance_id ] || {}, { deleted : 'deleted' } );
-					model.set( 'methods', methods );
-					model.logChanges( changes );
-					view.render();
+					shippingMethodView.block();
+
+					// Add method to zone via ajax call
+					$.post( {
+				    	url: ajaxurl + ( ajaxurl.indexOf( '?' ) > 0 ? '&' : '?') + 'action=woocommerce_shipping_zone_remove_method',
+						data: {
+							wc_shipping_zones_nonce: data.wc_shipping_zones_nonce,
+							instance_id: instance_id,
+							zone_id: data.zone_id,
+						},
+						success: function( { data } ) {
+							delete methods[instance_id];
+							changes.methods = changes.methods || data.methods;
+							model.set('methods', methods);
+							model.logChanges( changes );
+							view.render();
+							shippingMethodView.unblock();
+						},
+						error: function( jqXHR, textStatus, errorThrown ) {
+							console.error( 'There was an error removing the shipping method.', errorThrown );
```

Comment:
Instead of a `console.error` I think we should trigger a `window.alert` similar to other places within this `JS` file.
See `window.alert( data.strings.save_failed );` cases. We did have to add a new localized string, it looks like there is already one for `add_method_failed`, so we could also create one for `remove_method_failed`?

Reply:
Good call @louwie17! I just added the changes you mentioned. Could you take another look at this PR?

Reply:
Please scratch my comment. I'll add a couple of e2e tests before pinging you for a re review.